spec/system/community/community_spec.rb

```diff
@@ -791,6 +806,83 @@
       expect(page).to have_text(topic_2.title)
     end
 
+    scenario 'user searches for topic by title and apply filter topics with or without solution ' do
+      sign_in_user(coach.user, referrer: community_path(community))
+
+      topic_1.update!(title: 'Complex sentence with certain words')
+      topic_2.update!(title: 'Completely Different Sequence')
+
+      fill_in 'filter', with: 'complex sentence'
+
+      click_button 'Search by title: complex sentence'
+
+      fill_in 'filter', with: 'solution'
+
+      click_button 'Solution: Solved'
+
+      expect(page).to_not have_text(topic_2.title)
+      expect(page).to have_text(topic_1.title)
+      # Clear the filter
+      find("button[title='Remove selection: Solved']").click
+      find("button[title='Remove selection: complex sentence']").click
+
+      fill_in 'filter', with: 'Completely Different'
+
+      click_button 'Search by title: Completely Different'
+
+      fill_in 'filter', with: 'solution'
+
+      click_button 'Solution: Unsolved'
+
+      expect(page).to_not have_text(topic_1.title)
+      expect(page).to have_text(topic_2.title)
+    end
+
+    scenario 'user searches for topic by post body and apply filter topics with or without solution ' do
+      sign_in_user(coach.user, referrer: community_path(community))
+
+      topic_1.update!(title: 'Complex sentence with certain words')
+      topic_2.update!(title: 'Completely Different Sequence')
+      create :post,
+           topic: topic_1,
+           creator: student_1.user,
+           post_number: 4,
+           solution: true,
+           body: 'Another Complex Sentence'
+      create :post,
+           topic: topic_2,
+           creator: student_1.user,
+           post_number: 3,
+           solution: false,
+           body: 'Completely Different'
+
+
+      fill_in 'filter', with: 'complex sentence'
+
+      click_button 'Search by content: complex sentence'
+
+      fill_in 'filter', with: 'solution'
+
+      click_button 'Solution: Solved'
+
+      expect(page).to_not have_text(topic_2.title)
+      expect(page).to have_text(topic_1.title)
+      # Clear the filter
+      find("button[title='Remove selection: Solved']").click
+      find("button[title='Remove selection: complex sentence']").click
+
+      fill_in 'filter', with: 'Completely Different'
+
+      click_button 'Search by content: Completely Different'
+
+      fill_in 'filter', with: 'solution'
+
+      click_button 'Solution: Unsolved'
+
+      expect(page).to_not have_text(topic_1.title)
+      expect(page).to have_text(topic_2.title)
+    end
```

Comment:
I'm uncertain why these additional tests have been added. Did you encounter a bug when using these two filters together?

If so, there isn't actually a need to write a completely different test tests these two together. We can update the previous tests slightly to first test one filter, and then go on to engage both filters together, This sort of _composite_ tests can save a lot of testing time.